api/v1alpha1/nodehealthcheck_types.go

```diff
@@ -57,42 +57,60 @@ const (
 type NodeHealthCheckSpec struct {
 	// Label selector to match nodes whose health will be exercised.
 	// Note: An empty selector will match all nodes.
-	// +optional
+	//
+	//+optional
 	//+operator-sdk:csv:customresourcedefinitions:type=spec
 	Selector metav1.LabelSelector `json:"selector"`
 
 	// UnhealthyConditions contains a list of the conditions that determine
 	// whether a node is considered unhealthy.  The conditions are combined in a
 	// logical OR, i.e. if any of the conditions is met, the node is unhealthy.
 	//
-	// +optional
-	// +kubebuilder:default:={{type:Ready,status:False,duration:"300s"},{type:Ready,status:Unknown,duration:"300s"}}
-	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	//+optional
+	//+patchStrategy=merge
+	//+patchMergeKey=type
+	//+kubebuilder:default:={{type:Ready,status:False,duration:"300s"},{type:Ready,status:Unknown,duration:"300s"}}
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
 	UnhealthyConditions []UnhealthyCondition `json:"unhealthyConditions,omitempty"`
 
 	// Remediation is allowed if at least "MinHealthy" nodes selected by "selector" are healthy.
 	// Expects either a positive integer value or a percentage value.
 	// Percentage values must be positive whole numbers and are capped at 100%.
 	// 100% is valid and will block all remediation.
-	// +kubebuilder:default="51%"
-	// +kubebuilder:validation:XIntOrString
-	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
-	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	//
+	//+kubebuilder:default="51%"
+	//+kubebuilder:validation:XIntOrString
+	//+kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
 	MinHealthy *intstr.IntOrString `json:"minHealthy,omitempty"`
 
 	// RemediationTemplate is a reference to a remediation template
 	// provided by an infrastructure provider.
 	//
 	// If a node needs remediation the controller will create an object from this template
 	// and then it should be picked up by a remediation provider.
+	//
+	// Mutually exclusive with EscalatingRemediations
+	//
+	//+optional
 	//+operator-sdk:csv:customresourcedefinitions:type=spec
-	RemediationTemplate *corev1.ObjectReference `json:"remediationTemplate"`
+	RemediationTemplate *corev1.ObjectReference `json:"remediationTemplate,omitempty"`
 
-	// PauseRequests will prevent any new remdiation to start, while in-flight remediations
+	// EscalatingRemediations contain a list of ordered remediation templates with a timeout.
+	// The remediation templates will be used one after another, until the unhealthy node
+	// gets healthy within the timeout of the currently processed remediation.
+	//
+	// Mutually exclusive with RemediationTemplate
+	//
+	//+optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	EscalatingRemediations []EscalatingRemediation `json:"escalatingRemediations,omitempty"`
```

Comment:
Might be worth a note about the order in this list being irrelevant and that they need to use the `EscalatingRemediation.order` field

Reply:
will do in a follow up